lib/src/common/cache/cache.dart

```diff
@@ -294,6 +372,37 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
+  /// Returns `true` and calls [callback] if there is a cached, unreleased
+  /// [TValue] associated with [id]; returns `false` otherwise.
+  ///
+  /// This does not perform a [get] or [getAsync], and as a result, will not
+  /// affect retention or removal of a [TIdentifier][TValue] pair from the cache.
+  /// If any callbacks are in fight on removal of [TIdentifier] [TValue] pair
+  /// [didRemove] will not event until callback has completed.
```

Comment:
This could probably be made clearer and its own paragraph.

```dart
///
/// Any [TIdentifier] [TValue] pair removals will wait for the Future returned by
/// the call to [callback] before emitting [didRemove] events.
```